fix: question order preventing "present page" from working

jxjj opened the pull request:
Resolves #945 

- adds test for moving questions and question ordering (replicates issue in #945)
- update the FolderController so that when questions are returned with folder data (`/api/chime/:chimeId/folder/:folderId?includeQuestions`), the `order` data is remapped to `$index + 1` so that the front-end always gets normalized questions ordering (consecutive order starting at 1).
- updates the `updateQuestion` method so that if a question is moved to a different folder:
  1. it question is added to the end of the folder's question list
  2. the original folder and destination folder question order's are renormalized after the update for good measure
- fixes question seeder. Previously, order was always assigned to 1, which prevent some issues from surfacing. Now, order will be randomized.

On dev for testing

cmcfadden reviewed (commented):
is this just belt and suspenders or do we know of other cases that can trigger this? I'd be inclined to just fix the known issue and not add the sorting and renumbering on every folder load, as that's something we do a lot, unless we think it comes up. Could write a query to fix the existing cases if needed.

jxjj commented:
Hmm... not sure. I can take a look, but that makes me wonder if it's better to handle the sorting and reordering client-side and not bother with re-norming at all. What do you think?

cmcfadden commented:
My inclination would be to try to do something logical with sort order when moving folders (just throw it at the end of the list) but otherwise have the client be tolerant of the order being potentially wrong.

jxjj commented:
yeah, that sounds right to me.

Updates:
- removed server-side re-normalization (but kept the part where a moved question would be added to end of folder list).
- client-side added an `index` prop to each question, which is the 0-based, consecutive ordering.
- Updated question's present button to use `question.index`.

LMK if this is better or any change requests. 

On dev.

cmcfadden commented:
seems good to me.
